src/ilcdlib/epd/dialect/environdec.py

```diff
@@ -28,8 +33,31 @@ class EnvirondecIlcdXmlEpdReader(IlcdEpdReader):
 
     _TIME_REPR_DESC_DELIMITER = "\r\n"
 
-    def get_url_attachment(self, lang: LangDef) -> str | None:
-        """Return URL attachment if exists."""
+    def get_epd_document_stream(self) -> IO[bytes] | None:
+        """
+        Download the EPD document from the Environdec API, if it is possible.
+
+        In the case of Environdec EPD, the EPD document is not stored in the ILCD XML file, but is retrieved from the
+        Environdec their API and public EPD library.
+
+        We will not use this method to get the EPD document link for OpenEPD,
+        because it is a responsibility of the client to get related documents.
+        """
+
+        link = self.get_url_attachment("en")
+        if not link:
+            return None
+        if not re.match(r"https://www.environdec.com/Detail/epd\d+", link):
```

Comment:
Does this `if` mean the same as #bookmark below?

Reply:
They are not the same. Depends on url looks, way how we extract PDF document is a bit different

Reply:
Updated and added comments why this flow is used